CPAC/func_preproc/func_preproc.py

```diff
@@ -501,6 +501,190 @@ def get_idx(in_files, stop_idx=None, start_idx=None):
+            kernel_shape="sphere",
+            kernel_size=6.0,
+            internal_datatype="char",
+        ),
+        name=f"skullstrip_first_dilate_{pipe_num}",
+    )
+
+    bet_mask = pe.Node(fsl.ApplyMask(), name=f"skullstrip_first_mask_{pipe_num}")
+
+    # Use AFNI's unifize for T2 constrast
+    unifize = pe.Node(
+        afni_utils.Unifize(
+            t2=True,
+            outputtype="NIFTI_GZ",
+            args="-clfrac 0.2 -rbt 18.3 65.0 90.0",
+            out_file="uni.nii.gz",
+        ),
+        name=f"unifize_{pipe_num}",
+    )
+
+    # Run ANFI's 3dAutomask to extract a refined brain mask
+    skullstrip_second_pass = pe.Node(
+        preprocess.Automask(dilate=1, outputtype="NIFTI_GZ"),
+        name=f"skullstrip_second_pass_{pipe_num}",
+    )
+
+    # Take intersection of both masks
+    combine_masks = pe.Node(
+        fsl.BinaryMaths(operation="mul"), name=f"combine_masks_{pipe_num}"
+    )
+
+    # Compute masked brain
+    apply_mask = pe.Node(fsl.ApplyMask(), name=f"extract_ref_brain_bold_{pipe_num}")
+
+    node, out = strat_pool.get_data(["motion-basefile"])
+
+    wf.connect(
+        [
+            (node, init_aff, [(out, "moving_image")]),
+            (node, map_brainmask, [(out, "reference_image")]),
+            (node, norm, [(out, "moving_image")]),
+            (init_aff, norm, [("output_transform", "initial_moving_transform")]),
+            (
+                norm,
+                map_brainmask,
+                [
+                    ("reverse_invert_flags", "invert_transform_flags"),
+                    ("reverse_transforms", "transforms"),
+                ],
+            ),
+            (map_brainmask, binarize_mask, [("output_image", "in_file")]),
+            (binarize_mask, pre_dilate, [("out_file", "in_file")]),
+            (pre_dilate, print_header, [("out_file", "image")]),
+            (print_header, set_direction, [("header", "direction")]),
+            (node, set_direction, [(out, "infile"), (out, "outfile")]),
+            (set_direction, n4_correct, [("outfile", "mask_image")]),
+            (node, n4_correct, [(out, "input_image")]),
+            (n4_correct, skullstrip_first_pass, [("output_image", "in_file")]),
+            (skullstrip_first_pass, bet_dilate, [("mask_file", "in_file")]),
+            (bet_dilate, bet_mask, [("out_file", "mask_file")]),
+            (skullstrip_first_pass, bet_mask, [("out_file", "in_file")]),
+            (bet_mask, unifize, [("out_file", "in_file")]),
+            (unifize, skullstrip_second_pass, [("out_file", "in_file")]),
+            (skullstrip_first_pass, combine_masks, [("mask_file", "in_file")]),
+            (skullstrip_second_pass, combine_masks, [("out_file", "operand_file")]),
+            (unifize, apply_mask, [("out_file", "in_file")]),
+            (combine_masks, apply_mask, [("out_file", "mask_file")]),
+        ]
+    )
+
+    outputs = {
+        "fMRIprep_brain_mask": (combine_masks, "out_file"),
```

Comment:
Please keep these in BIDS format, even though it's an intermediate naming from what I can tell.
Is there a reason not to also name this `space-bold_desc-brain_mask`?

Reply:
The output of the `combine_masks` node has always been the complete BOLD mask. I don't think there is a need to have it as a separate name if it will end up being re-written into `outputs` as `space-bold_desc-brain_mask`.

If we are running multiple bold masking strategies, then they all have to be the same standardized resource name. They will all be in the resource pool as `space-bold_desc-brain_mask` but each different strategy will be stored underneath that resource.

Reply:
Right, so, as long as the `coregistration_prep_fmriprep` nodeblock only outputs `sbref` in that final `outputs` dictionary (see thread further below), it won't fork, and there will only be the FSL bold mask in the resource pool. What gets included in `outputs` is the only thing that goes into the resource pool.